### PR TITLE
Gives lategame revs and crew revhead/head trackers

### PR DESCRIPTION
**Pull request by ThatFiveGuys**
27-30 minutes past roundstart on a rev round, a broadcast gets sent out to everyone that gives every pda both a program for tracking revheads, and a program for tracking heads. This tracker shows the direction the nearest head is in, and the distance of the nearest head. 

image

## About the PR 
Lategame revs can tend to drag on to become a game of side that has already practically won tracks down the single hidden/afk head remaining. This should give a climactic finish to rev rounds by making it a rush to defend/attack the remaining heads.

## Changelog

```changelog
(u)Ikea
(*)27-30 minutes into a rev round everyone recieves a tracker for revheads/heads on their pda.
```

### CODE REVIEW

**Comment by CaioFrancisco** [comment]
A few things that I wanted to happen: the PDA program could be spread to Revolutionaries and Command around the same time the first few gibwave announcements start, so CentComm doesn't warn about revs twice. Another thing I noticed is the poor grammar in the announcements, so I'll send an improved version here:

`A revolution has been detected on [station_name(1)]. A station-wide scan has revealed the causes of the revolution, and a PDA program has been sent to all the Heads of Staff.`

For the head revs, there could be a more silent way to indicate this by sending a PDA message only to them, like shown on your screenshot. But all of this is just my idea, if you want to put it in your code is up to you.

**Comment by ThatFiveGuys** [comment]
To me the point of the first gibwave message is mostly to say “wrap shit up shit is about to get fucked if you dont”, I picked 27-30 minutes into the round as thats when rev rounds feel like they should begin the third act. 

I dont want the tracker to be limited in who its sent to for a few reasons. First off its not unusual to have the last heads remaining afk and this could easily kill the momentum of a side in its tracks which seems bad. Finally, i want this to be the start of a proper conclusion for rev rounds. Currently rev rounds just kind of end abruptly after one side has already basically won and its several minutes of mostly non action, this sucks. The intent here is that by giving everyone the program at once, you cause everyone to swarm to defend/attack heads, this should cause a bunch of holdouts, lone wolf operations, and all around just a lot of action and chaos. I think limiting this program to only a few people would severely limit that.

**Comment by CaioFrancisco** [comment]
fair enough, the only complaint left is just the grammar on the announcements and what seems like a lack of explanation for a few things. PDA is an acronym (personal data assistant) and you probably meant "PDA's" as in plural. it's probably just me, but "station-wide" and "on-going" sound much better with the dashes, and the announcements would feel more appropriate if they had the same tone as when it's gangs and CentComm just wants shit to be done.

**Comment by Conchuckter** [comment]
Something to note is that if someones PDA is full then they will not receive it, this could be exploited if someone decided to spam file transfers to intentionally fill the crews PDA's.

**Comment by ThatFiveGuys** [comment]
I seriously doubt that will happen and will probably violate the rules due to it requiring spamming pdas with measages so im not too concerned

**Comment by ZeWaka** [comment]
> I seriously doubt that will happen and will probably violate the rules due to it requiring spamming pdas with measages so im not too concerned

this is pretty trivial and i wouldn't consider this against the rules on antag grief

File: code/datums/gamemodes/revolution.dm [hunk 1]
```diff
@@ -17,6 +17,8 @@
 	var/list/datum/mind/revolutionaries = list()
 	var/finished = 0
 	var/const/waittime_l = 600 //lower bound on time before intercept arrives (in tenths of seconds)
+	var/const/TrackerTime_min = 27 MINUTES //upper bound on time before intercept arrives (in tenths of seconds)
+	var/const/TrackerTime_max = 30 MINUTES //lower bound on time before intercept arrives (in tenths of seconds)
```
Comment (Conchuckter) [inline code/datums/gamemodes/revolution.dm:21]: Incorrect comments, also looks like you placed it between two corresponding variables.

**Review by Conchuckter — CHANGES_REQUESTED** [review]
The annoucements need some work as others have said, stuff like pda needs to be in uppercase.

**Comment by ThatFiveGuys** [comment]
> fair enough, the only complaint left is just the grammar on the announcements and what seems like a lack of explanation for a few things. PDA is an acronym (personal data assistant) and you probably meant "PDA's" as in plural. it's probably just me, but "station-wide" and "on-going" sound much better with the dashes, the announcements would feel more appropriate if they had the same tone as when it's gangs and CentComm just wants shit to be done and the initial announcement would be better if it already told you that it's a revolution, not a foreign mutiny.

Forgot to respond for this, I chose a very clinical tone for centcomm as I thought itd be most fitting, and I wanted the two announcements to be distinct. The reason why I used foreign mutiny first is because using the term revolution didnt seem fitting for Centcomm to describe the round as. Im not worried about clarity here as the fact that its a rev round should be very clear and I used the term rev leader specifically to specify what the tracker does. I dont think the dashes look as good

I didnt want to specify entirely what was going on because I think it fits most for you have to put a few things together to get the full picture. Since the syndicate doesn't have full access to the station, the syndicate broadcast is piggybacked ontop of centcomm sending the tracker. The pda isn't the thing actually tracking the heads lore wise, its the sensor arrays, hence why the syndicate has to identify the biometric signatures of command and not just send it out instantly. The reason why the gibwave comes so late is that just in case, nanotrasen wants to truly confirm that the signatures they had for the revheads are correct (hence the "The Revolutionary heads' biometric signatures have been confirmed.")

**Comment by ThatFiveGuys** [comment]
Sending files to pdas through file_send on pdas now requires a new netpass_centcom, meaning you can no longer spam pda's with files unless you manage to sniff the message that sends out the tracker message.

**Comment by Conchuckter** [comment]
> Sending files to pdas through file_send on pdas now requires a new netpass_centcom, meaning you can no longer spam pda's with files unless you manage to sniff the message that sends out the tracker message.



> Sending files to pdas through file_send on pdas now requires a new netpass_centcom, meaning you can no longer spam pda's with files unless you manage to sniff the message that sends out the tracker message.

This breaks a ton of stuff, and using `batt_adjust` makes no sense. You could instead set aside some storage space that can only be written to when `netpass_centcom` is present, or you could send a `file_send_req` and send the file to each PDA only when they reply with `file_send_acc`.

**Comment by ThatFiveGuys** [comment]
I just used batt+adjust because its what detomatix uses and I dont know packet stuff tbh

**Comment by ThatFiveGuys** [comment]
Just decided on setting the file sizes to zero

**Comment by ZWRh3** [comment]
i agree that revs would benefit from mass trackers as once one side is loosing it turns into pod/locker hide and seek rather than revs just today half the heads delayed the round by around 20 minutes by hiding in lockers

**Comment by ZeWaka** [comment]
Let's try this out.

**Pull request merged**